src/cmd/subscribe.js

```diff
@@ -1,77 +1,90 @@
+const os = require('os');
 const VError = require('verror');
-const ApiClient = require('../lib/api-client');
+const settings = require('../../settings');
+const spinnerMixin = require('../lib/spinner-mixin');
+const { normalizedApiError } = require('../lib/api-client');
+const { errors: { usageError } } = require('../app/command-processor');
+const ParticleAPI = require('./api');
+const UI = require('../lib/ui');
 
 
 module.exports = class SubscribeCommand {
-	startListening(event, { device, all }) {
-		const api = new ApiClient();
-		api.ensureToken();
+	constructor({
+		stdin = process.stdin,
+		stdout = process.stdout,
+		stderr = process.stderr
+	} = {}){
+		this.stdin = stdin;
+		this.stdout = stdout;
+		this.stderr = stderr;
+		this.ui = new UI({ stdin, stdout, stderr });
+		spinnerMixin(this);
+	}
 
+	startListening({ device, all, product, params: { event } }){
 		// legacy argument order
 		let eventName = event[0];
 		let deviceId = event[1] || device;
 
 		// if they typed: "particle subscribe mine"
-		if (eventName === 'mine') {
+		if (eventName === 'mine'){
 			eventName = null;
 		}
 
 		let eventLabel = eventName;
-		if (eventLabel) {
+		if (eventLabel){
 			eventLabel = '"' + eventLabel + '"';
 		} else {
 			eventLabel = 'all events';
 		}
 
-		if (!deviceId && !all) {
+		if (!deviceId && !all){
 			deviceId = 'mine';
 		}
 
-		if (!deviceId) {
-			console.log('Subscribing to ' + eventLabel + ' from the firehose (all devices) and my personal stream (my devices)');
-		} else if (deviceId === 'mine') {
-			console.log('Subscribing to ' + eventLabel + ' from my personal stream (my devices only) ');
-		} else {
-			console.log('Subscribing to ' + eventLabel + ' from ' + deviceId + "'s stream");
-		}
-
-		let chunks = [];
-		function appendToQueue(arr) {
-			for (let i = 0; i < arr.length; i++) {
-				const line = (arr[i] || '').trim();
-				if (!line) {
-					continue;
-				}
-				chunks.push(line);
-				if (line.indexOf('data:') === 0) {
-					processItem(chunks);
-					chunks = [];
-				}
+		if (product){
+			if (!device){
+				throw usageError(
```

Comment:
It's possible to subscribe all product events, not just product events from a specific device.

Reply:
✅ done in https://github.com/particle-iot/particle-cli/pull/558/commits/ae3d983343e4e562175efc2587241e5e8f6062ec